lib/push_2_tweet_server.js

```diff
@@ -0,0 +1,316 @@
+ * @return {Object} The context to be used for logging
+ */
+function getContext(request) {
+  return {
+    corr: request.headers[p2tConfig.UNICA_CORRELATOR_HEADER] ||
+            p2tHelper.getUnicaCorrelator(request),
+    trans: p2tHelper.getTransactionId(),
+    op: p2tHelper.getOperationType(request)
+  };
+}
+
+/**
+ * Attends a received request
+ * @param request The received request
+ * @param callback Callback to be notified when the request has been processed
+ */
+function processOperationRequest(request, callback) {
+  twitter.post(
+    'statuses/update',
+    {
+      status: '(TESTING) Button with id: \'' + request.push2Tweet.operationDescriptor.button + '\'' +
+      ', action: \'' + request.push2Tweet.operationDescriptor.action + '\'' +
+      ' and extra: \'' + request.push2Tweet.operationDescriptor.extra + '\' has just been pushed!'
+    },
+    function(err, tweet, response){
+      if (err && callback) {
+        return callback(err);
+      } else {
+        p2tLogger.info(request.push2Tweet.context, 'Tweet published: \'' + tweet.text + '\'');
+        request.push2Tweet.result = {
+          code: 'TWEET_SUCCESSFULLY_PUBLISHED',
+          tweet: tweet,
+          response: response
+        };
+        if (callback) {
+          return callback();
+        }
+      }
+    }
+  );
+}
+
+/**
+ * Error handler in case an error occurs during the request processing
+ * @param request The received request
+ * @param reply hapi's reply() function
+ * @param err The error which occurred
+ */
+function requestErrorHandler(request, reply, err) {
+  if (err) {
+    p2tComm.respond2Request(err, request, reply);
+  }
+}
+
+/**
+ * Handler to manage requests
+ * @param {Object} request The received request
+ * @param {Function} reply The reply function to respond to the requester
+ */
+function requestHandler(request, reply) {
+  request.push2Tweet = request.push2Tweet || {};
+  request.push2Tweet.context = getContext(request);
+
+  p2tLogger.debug(
+    request.push2Tweet.context,
+    'new request received: ' +
+      JSON.stringify(request.payload)
+  );
+
+  async.waterfall([
+    p2tComm.getOperationDescriptor.bind(null, request),
+    processOperationRequest.bind(null, request),
+    p2tComm.respond2Request.bind(null, null, request, reply)
+  ], requestErrorHandler.bind(null, request, reply));
+}
+
+/**
+ * Starts the server asynchronously
+ * @param {string} host The Push 2 Tweet server host
+ * @param {string} port The Push 2 Tweet server port
+ * @param {Function} callback Callback function to notify the result
+ *  of the operation
+ */
+function start(host, port, callback) {
+  var context = {
+    op: p2tConfig.OPERATION_TYPE.SERVER_LOG
+  };
+
+  server = new hapi.Server();
+
+  server.on('log', function(event, tags) {
+    if (tags.load) {
+      p2tLogger.warn(context, 'event=' + JSON.stringify(event));
+    }
+  });
+
+  server.on('request-internal', function(request, event, tags) {
```

Comment:
Fixed in https://github.com/telefonicaid/push-2-tweet/commit/924d17e089f7e52b972e5408ec4dd5d8f45c8522